From 382366c1f615613e088982a6c30fc90b3226ae8b Mon Sep 17 00:00:00 2001
From: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
Date: Fri, 20 Nov 2020 12:14:31 +0000
Subject: [PATCH 2/2] ucm2: add support to for Qualcomm RB5 Platform

The Qualcomm RB5 Robotics Platform contains HDMI,
2x WSA Smart-Speakers audio outputs along with One
Onboard DMIC audio input.

Signed-off-by: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
---
 ucm2/sm8250/HDMI.conf                         | 26 +++++++++++++
 ucm2/sm8250/HiFi.conf                         | 37 +++++++++++++++++++
 .../Qualcomm-RB5-WSA8815-Speakers-DMIC0.conf  | 11 ++++++
 3 files changed, 74 insertions(+)
 create mode 100644 ucm2/sm8250/HDMI.conf
 create mode 100644 ucm2/sm8250/HiFi.conf
 create mode 100644 ucm2/sm8250/Qualcomm-RB5-WSA8815-Speakers-DMIC0.conf

diff --git a/ucm2/sm8250/HDMI.conf b/ucm2/sm8250/HDMI.conf
new file mode 100644
index 000000000000..a9594fd94af2
--- /dev/null
+++ b/ucm2/sm8250/HDMI.conf
@@ -0,0 +1,26 @@
+# Use case configuration for RB5 board.
+# Author: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
+
+SectionVerb {
+	EnableSequence [
+		cset "name='TERT_MI2S_RX Audio Mixer MultiMedia1' 1"
+	]
+
+	DisableSequence [
+		cset "name='TERT_MI2S_RX Audio Mixer MultiMedia1' 0"
+	]
+
+	Value {
+		TQ "HiFi"
+	}
+}
+
+SectionDevice."HDMI" {
+	#Name "HDMI"
+	Comment "HDMI Digital Stereo Output"
+
+	Value {
+		PlaybackPCM "hw:${CardId}"
+		PlaybackPriority 200
+	}
+}
diff --git a/ucm2/sm8250/HiFi.conf b/ucm2/sm8250/HiFi.conf
new file mode 100644
index 000000000000..484bb49057e7
--- /dev/null
+++ b/ucm2/sm8250/HiFi.conf
@@ -0,0 +1,37 @@
+# Use case configuration for Qualcomm RB5.
+# Author: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
+
+SectionVerb {
+
+	EnableSequence [
+		cset "name='WSA_CODEC_DMA_RX_0 Audio Mixer MultiMedia2' 1"
+		<codecs/wsa881x/DefaultEnableSeq.conf>
+	]
+
+	DisableSequence [
+		cset "name='WSA_CODEC_DMA_RX_0 Audio Mixer MultiMedia2' 0"
+	]
+
+	Value {
+		TQ "HiFi"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker playback"
+
+	EnableSequence [
+		<codecs/lpass/wsa-macro/SpeakerEnableSeq.conf>
+		<codecs/wsa881x/SpeakerEnableSeq.conf>
+	]
+
+	DisableSequence [
+		<codecs/wsa881x/SpeakerDisableSeq.conf>
+		<codecs/lpass/wsa-macro/SpeakerDisableSeq.conf>
+	]
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId},1"
+	}
+}
diff --git a/ucm2/sm8250/Qualcomm-RB5-WSA8815-Speakers-DMIC0.conf b/ucm2/sm8250/Qualcomm-RB5-WSA8815-Speakers-DMIC0.conf
new file mode 100644
index 000000000000..be2aea7aab22
--- /dev/null
+++ b/ucm2/sm8250/Qualcomm-RB5-WSA8815-Speakers-DMIC0.conf
@@ -0,0 +1,11 @@
+Syntax 3
+
+SectionUseCase."HiFi" {
+	File "/sm8250/HiFi.conf"
+	Comment "HiFi quality Music."
+}
+
+SectionUseCase."HDMI" {
+	File "/sm8250/HDMI.conf"
+	Comment "HDMI output."
+}
-- 
2.29.2

